modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -363,32 +363,21 @@ function editConsentStatusFields($db, $user)
         exit;
     }
 
+    $currentUser = User::singleton();
+    $uid         = $currentUser->getData("UserID");
+
     $candIDParam = $_POST['candID'];
     $candID      = (isset($candIDParam) && $candIDParam !== "null") ?
         $candIDParam : null;
 
-    $id = null;
-    if (!(is_null($_SESSION['State']))) {
-        $currentUser =& User::singleton($_SESSION['State']->getUsername());
-        $id          = $currentUser->getData("UserID");
-    }
+    $consentDetails   = Utility::getConsentList();
+    $candidateConsent = Candidate::getConsent($candID);
```

Comment:
It's silly to have a static function in the candidate class which takes the identifier for a candidate as a parameter and returns an attribute of that candidate. That's pretty much the definition of what should be a (non-static) method on a candidate object.

I also don't think we should be adding new methods to Utility, we should be trying to move functions out of there to places that make more sense (in this case, a new Consent or ConsentTypes class would probably make the most sense.)